### PR TITLE
fix: trim input filename and test name

### PR DESCRIPTION
**Pull request by btea**

### CODE REVIEW

File: packages/vitest/src/node/core.ts [hunk 1]
```diff
@@ -364,13 +364,13 @@ export class Vitest {
   }
 
   async changeNamePattern(pattern: string, files: string[] = this.state.getFilepaths(), trigger?: string) {
-    this.config.testNamePattern = pattern ? new RegExp(pattern) : undefined
+    this.config.testNamePattern = pattern.trim() ? new RegExp(pattern) : undefined
```
Comment (sheremet-va) [inline packages/vitest/src/node/core.ts:367]: Shouldn't trimmed patter be used in `new RegExp`? Also maybe it's better to trim it before `changeNamePattern` is called? The same for `changeFilenamePattern`

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/node/core.ts (hunk 1 above)
Comment (btea) [reply packages/vitest/src/node/core.ts:367]: Maybe here?
https://github.com/vitest-dev/vitest/blob/d6225764856dd62cb1b21ec5bf4619052609c244/packages/vitest/src/node/stdin.ts#L98-L99

Comment (sheremet-va) [reply packages/vitest/src/node/core.ts:367]: Yes

Comment (btea) [reply packages/vitest/src/node/core.ts]: Updated.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**